var/spack/repos/builtin/packages/python/package.py

```diff
@@ -148,9 +153,16 @@ def configure_args(self):
             'CPPFLAGS=-I{0}'.format(' -I'.join(dp.include for dp in dep_pfxs)),
             'LDFLAGS=-L{0}'.format(' -L'.join(dp.lib for dp in dep_pfxs)),
         ]
+
+        if '@2.7.13:2.8,3.5.3:' in spec and '+optimizations' in spec:
+            config_args.append('--enable-optimizations')
+
         if spec.satisfies('%gcc platform=darwin'):
             config_args.append('--disable-toolbox-glue')
 
+        if '%intel' in spec and '@2.7.12:2.8,3.5.2:' in spec:
```

Comment:
just for my understanding: `'@2.7.12:2.8,3.5.2:' in spec` means that the version of this particular package is required to be in that range. Intuitively I would have expected that this means that ANY package in the spec (i.e. also some dependency) satisfies this requirement

Reply:
> Intuitively I would have expected that this means that ANY package in the spec (i.e. also some dependency) satisfies this requirement

That's technically true, see #4135 . For convenience the current implementation is:
```python
def __contains__(self, spec):
    """True if this spec satisfies the provided spec, or if any dependency
       does.  If the spec has no name, then we parse this one first.
    """
    spec = self._autospec(spec)
    for s in self.traverse():
        if s.satisfies(spec, strict=True):
            return True

    return False
```
On the other hand it's a commonly used idiom in packages, instead of:
```python
spec.satisfies('@2.7.12:2.8,3.5.2:', strict=True)
```
probably because the latter is heavier in syntax and requires you to konw what `strict` means. Note that `strict=True` is not strictly needed for versions alone, but if you forget it and later on you put a mv variant into the mix things will start to behave in an unexpected way. 

In any case, if you prefer this changed let me know. I don't have strong opinions on it.

Reply:
Long term we should definitely straighten this syntax out... I'm not familiar enough with the Spec-implementation to hazard a guess how hard it will be to do that though.

But as `'@2.7.12:2.8,3.5.2:' in spec` really doesn't mean "if the python version is new enough" I'd prefer `spec.satisfies`

Reply:
Ok, the logic is now correct (but the syntax looks ugly)